docs/using_onemkl_with_cmake.rst

```diff
@@ -0,0 +1,102 @@
+.. _using_onemkl_interface_library_with_cmake:
+
+Using the oneMKL Interfaces in your project with CMake
+=============================================================
```

Comment:
I've seen warnings when rendering previous .rst doc when the title symbols are not as long as the text above. Do you have any warnings when rendering the doc? There are a few other occurrences.
I could use another html attachment to review the recent changes if that's ok!

Reply:
No warnings at all. `cmake --build . --verbose` gives
```
Build succeeded.
    0 Warning(s)
    0 Error(s)
```
[html.zip](https://github.com/user-attachments/files/15884579/html.zip)

Reply:
Right, I thought this was due to the `-Q` flag but even without it I don't see any warning. The output looks good to me.